neuromation/api/quota.py

```diff
@@ -0,0 +1,59 @@
+from dataclasses import dataclass
+from typing import Any, Dict, Optional
+
+from yarl import URL
+
+from neuromation.api.config import _Config
+from neuromation.api.core import _Core
+from neuromation.api.utils import NoPublicConstructor
+
+
+@dataclass(frozen=True)
+class QuotaDetails:
+    spent_minutes: int
+    limit_minutes: Optional[int]
+
+    @property
+    def remain_minutes(self) -> Optional[int]:
+        if self.limit_minutes is None:
+            # remain: infinity
+            return None
+        if self.limit_minutes > self.spent_minutes:
+            return self.limit_minutes - self.spent_minutes
+        return 0
+
+
+@dataclass(frozen=True)
+class QuotaInfo:
+    name: str
+    gpu_details: QuotaDetails
+    cpu_details: QuotaDetails
+
+
+class Quota(metaclass=NoPublicConstructor):
+    def __init__(self, core: _Core, config: _Config) -> None:
+        self._core = core
+        self._config = config
+
+    async def get(self) -> QuotaInfo:
```

Comment:
How to get quota for another user?
I can do it if have appropriate permissions, right?

Reply:
do we really need a funcitonality to see other user's quota? cc @mariyadavydova

Reply:
We obviously support it by REST API. Otherwise please modify the server as well.

Reply:
Ideally, yes, we need a functionality to see other user's quota (for example, I would like to be able to see the quotas left for our free tier users). However, `config` seems to be a bad place for such functionality, as other `config` commands refer to your personal information (or your server's). Let's file another issue and think where we should put the quota-related functionality (this refers to other discussions in this PR too). This particular issue should solve the immediate problem - "How do I, free tier user, get to know how much quota I have?".

Reply:
> We obviously support it by REST API. Otherwise please modify the server as well.

Please, don't :) I use this functionality to bump quotas via REST API.

Reply:
@mariyadavydova here is a public API class, not `neuro config` CLI command group.
API is more generic (and potentially harder to change publishing).

Reply:
Oh, I'm sorry. Then I think that we should update this class accordingly and keep it as is for a while, as this quota functionality is emerging now and we do not have a clear understanding of how it should be organized.

Reply:
`async def get(self, user: Optional[str]=None)`, the default `None` is for self.

Reply:
already done